chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/certifiedProduct/AmbulatoryModular2014Validator.java

```diff
@@ -91,8 +93,25 @@ protected void validateDemographics(PendingCertifiedProductDTO product) {
 							product.getErrorMessages().add("There was no test tool found matching '" + testTool.getName() + "' for certification " + cert.getNumber() + ".");
 						} else {
 							TestToolDTO tt = ttDao.getById(testTool.getTestToolId());
-							if(tt != null && tt.isRetired()) {
-								product.getErrorMessages().add("Cannot use test tool '" + tt.getName() + "' since it is retired. Please choose a different test tool for " + cert.getNumber() + ".");
+							String[] idParts = product.getUniqueId().split("\\.");
+							if(idParts.length < 9) {
+								logger.error("CHPL ID must have 9 parts separated by '.'");
+							}
+							if(tt != null && tt.isRetired() && idParts[6].toString().equals("0")) {

```

Comment:
Looks like if the id doesn't have 9 parts it logs an error but then continues on trying to access the 6th part. Should this be in an else or reorganized?